docs/madr/decisions/005-policy-matching.md

```diff
@@ -0,0 +1,1052 @@
+# New Policy Matching
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/3330 
+
+## Context and Problem Statement
+
+Source and destination selectors (S&D) are trying to solve the problem of mapping between policies and connections. 
+But in the real world of Envoy proxies, 
+there is no such thing as "connection". 
+There are "inbounds" and "outbounds". 
```

Comment:
nitpick: Envoy does not mention "inbounds" or "outbounds" in the [terminology docs](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/intro/terminology) - there are just `Listeners` with [TrafficDirection](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#envoy-v3-api-enum-config-core-v3-trafficdirection) set to `INBOUND` or `OUTBOUND`. I think that the intended audience will figure it out but for clarity sake we could define the terms "inbounds" and "outbounds".

Reply:
Yeah, inbounds and outbounds are something that Kuma is using when configuring Envoy. We defined these terms in [doc](https://kuma.io/docs/dev/explore/dpp/#concepts), but it's still under `dev`. Since this is a MADR that is targeted at developers, I think it's fine not to define inbound and outbound in this doc.